docs/framework/dataframe.mdx

```diff
@@ -2,97 +2,213 @@
-
-| compatibility      | dataframe                             |
-|--------------------|---------------------------------------|
-| `pandas.DataFrame` | x                                     |
-| `polar.DataFrame`  | x                                     |
-| `list of records`  | x (with `EditableDataframe`)          |
-
-## Use a dataframe
-
-**a dataframe is simply added to the state**. A component like `dataframe` will be able to display it.
+**Writer Framework has two objects with the name _DataFrame_:**
+
+1. **Code Dataframe:** In code that you write for a Writer Framework application, `DataFrame` refers to a **_data structure_** that stores data in rows and columns in a way similar to a spreadsheet or SQL table.
+2. **UI DataFrame:** In Writer Framework's UI, "DataFrame" refers to a **_user interface component_** that displays data in rows and columns in a way similar to a spreadsheet or SQL table. 
+
+To present a data table in Writer Framework, you create a `DataFrame` data structure in your code and then bind it to a UI Dataframe.
+
+
+## Displaying a static DataFrame
+
+A static DataFrame is one whose content does not change. The user can change its sort order, but the data within the DataFrame remains constant.
+
+<Steps>
+	<Step title="Create a DataFrame data structure">
+		Writer Framework supports both [pandas](https://pandas.pydata.org/) and [Polars](https://pola.rs/) `DataFrame` data structures. Create a `DataFrame`, assign its value to a variable, then assign make that variable a value in the `state` dictionary:
+
+		<CodeGroup>
+			```python pandas
+			import writer as wf
+			import pandas as pd
+
+			data = [
+				{"rank": 1, "flavor": "Vanilla", "favorite": 0.11},
+				{"rank": 2, "flavor": "Chocolate", "favorite": 0.1},
+				{"rank": 3, "flavor": "Cookies and cream", "favorite": 0.07},
+				{"rank": 4, "flavor": "Strawberry", "favorite": 0.06},
+				{"rank": 5, "flavor": "Chocolate chip", "favorite": 0.02},
+			]
+			df = pd.DataFrame(data)
+
+			wf.init_state({
+				"mydf": df
+			})
+			```
+
+			```python Polars
+			import writer as wf
+			import polars as pl
+
+			data = [
+				{"rank": 1, "flavor": "Vanilla", "favorite": 0.11},
+				{"rank": 2, "flavor": "Chocolate", "favorite": 0.1},
+				{"rank": 3, "flavor": "Cookies and cream", "favorite": 0.07},
+				{"rank": 4, "flavor": "Strawberry", "favorite": 0.06},
+				{"rank": 5, "flavor": "Chocolate chip", "favorite": 0.02},
+			]
+			df = pl.DataFrame(data)
+
+			wf.init_state({
+				"mydf": df
+			})
+			```
+		</CodeGroup>
+
+		The call to `wf.init_state()` adds the `DataFrame` to the application's `state` variable as the value of the `mydf` key. 
+	</Step>
+	<Step title="Add a DataFrame component to the UI and bind it to the DataFrame data structure">
+		Add a DataFrame UI component to the user interface, then set its **Data** property to `@{`_dataframe_key_`}`, where _dataframe_key_ is the `state` variable key whose value refers to the `DataFrame` data structure.
```

Comment:
The underscores at the front and end of `_dataframe_key_` add italics and break the code formatting, I'd remove them.